sound/soc/sof/amd/renoir.c

```diff
@@ -16,6 +16,94 @@
 #include <sound/sof.h>
 
 #include "acp.h"
+#include "acp-dsp-offset.h"
+
+#define I2S_BT_INSTANCE		0
+#define I2S_SP_INSTANCE		1
+#define PDM_DMIC_INSTANCE	2
+
+#define I2S_MODE		0x04
+
+static int renoir_dai_probe(struct snd_soc_dai *dai)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(dai->component);
+	unsigned int val;
+
+	snd_sof_dsp_write(sdev, ACP_DSP_BAR, ACP_I2S_PIN_CONFIG, I2S_MODE);
+
+	val = snd_sof_dsp_read(sdev, ACP_DSP_BAR, ACP_I2S_PIN_CONFIG);
+	if (val != I2S_MODE) {
+		dev_err(sdev->dev, "I2S Mode not supported val %x\n", val);
```

Comment:
the error wording is misleading isnt it? from the check, it looks like I2S_MODE is the only supported mode. But the error says "I2S_Mode not supported"?

Reply:
Looks like a readback test to me as well. Probably the print should be something like:
`"I2S mode is not supported (I2S_PIN_CONFIG: %#x", val` ?